apollo-router/src/plugins/connectors/handle_responses.rs

```diff
@@ -488,20 +490,17 @@ pub(crate) async fn process_response<T: HttpBody>(
     debug_context: &Option<Arc<Mutex<ConnectorContext>>>,
     supergraph_request: Arc<http::Request<crate::graphql::Request>>,
 ) -> connector::request_service::Response {
-    match result {
+    let (mapped_response, result) = match result {
         // This occurs when we short-circuit the request when over the limit
         Err(error) => {
             let raw = RawResponse::Error {
-                error: error.to_graphql_error(
-                    connector.clone(),
-                    Some(Path::from_response_key(&response_key)),
-                ),
+                error: error.to_runtime_error(&connector, response_key.clone()),
```

Comment:
```suggestion
                error: error.to_runtime_error(&connector, &response_key),
```
With another change in another comment